incubator/hnc/pkg/reconcilers/hierarchy_config.go

```diff
@@ -89,20 +84,22 @@ func (r *HierarchyConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 	ctx := context.Background()
 	ns := req.NamespacedName.Namespace
 
+	// Todo this will be removed soon once HNSR is updated to not enqueue empty
+	// 	namespace to HCR.
+	if ns == "" {
+		return ctrl.Result{}, nil
+	}
+
 	rid := (int)(atomic.AddInt32(&r.reconcileID, 1))
 	log := r.Log.WithValues("ns", ns, "rid", rid)
 
-	// TODO remove this log and use the HNSReconcilerEnabled to toggle the behavour of this
-	//  reconciler accordingly. See issue: https://github.com/kubernetes-sigs/multi-tenancy/issues/467
-	// Output a log for testing.
-	log.Info("HC will be reconciled with", "HNSReconcilerEnabled", r.HNSReconcilerEnabled)
-
 	return ctrl.Result{}, r.reconcile(ctx, log, ns)
 }
 
 func (r *HierarchyConfigReconciler) reconcile(ctx context.Context, log logr.Logger, nm string) error {
 	// Get the singleton and namespace.
 	inst, nsInst, err := r.GetInstances(ctx, log, nm)
+	hnsnms, err := r.getHierarchicalNamespaceNames(ctx, nm)
```

Comment:
Should we add a watch on the HNSes as well?

Reply:
Also: if we add these to the forest (eg `ns.SetHNSes(hnsnms)`), then it's easy to set and clear the `HNS_MISSING` condition on the child here in the HCR. All we need to do is re-invoke the HCR on any children when the HNS is added or removed, eg something like:

```
for changedHNS := range ns.SetHNSes(hnsNms) {
  r.enqueueAffected(changedHNS)
}
```
Make sense?

Reply:
yes, good idea!